test/spmd/tensor/test_matrix_ops.py

```diff
@@ -1,31 +1,84 @@
 # Copyright (c) Meta Platforms, Inc. and affiliates
 import torch
+from torch.distributed.distributed_c10d import ReduceOp
 from torch.testing._internal.common_utils import run_tests
 from ..test_utils import DistTensorTestBase, with_comms
-from spmd import distribute_tensor, DeviceMesh, Shard, Replicate
+from spmd import (
+    distribute_tensor,
+    DeviceMesh,
+    Shard,
+    Replicate,
+    _Partial,
+    Tensor,
+)
 
 
 class DistMatrixOpsTest(DistTensorTestBase):
     @with_comms
     def test_addmm(self):
-        device_mesh = DeviceMesh(self.device_type, list(range(self.world_size)))
-        shard_spec = [Shard(0)]
-        replica_spec = [Replicate()]
+        from torch.distributed import get_rank
 
-        tensor_to_shard = torch.randn(12, 8)
-        mat1 = distribute_tensor(tensor_to_shard, device_mesh, shard_spec)
-        tensor_to_replicate = torch.randn(8, 4)
-        mat2 = distribute_tensor(tensor_to_replicate, device_mesh, replica_spec)
-        input_tensor = torch.randn(4)
-        input = distribute_tensor(input_tensor, device_mesh, replica_spec)
+        rank = get_rank()
+        world_size = self.world_size
+
+        device_mesh = DeviceMesh(self.device_type, list(range(world_size)))
+        placement_rep = [Replicate()]
+        placement_par = [_Partial(ReduceOp.SUM)]
+        placement_s0 = [Shard(0)]
+        placement_s1 = [Shard(1)]
+
+        # Ground truth
+        torch.manual_seed(0)
+        input = torch.randn((world_size, world_size))
+        mat1 = torch.randn((world_size, world_size * 2))
```

Comment:
also as Junjie suggested, we need to directly test on `nn.Linear` as well in additional to addmm